bundle/src/main/java/com/adobe/acs/commons/hc/impl/SMTPMailServiceHealthCheck.java

```diff
@@ -0,0 +1,117 @@
+                name = HealthCheck.TAGS,
+                unbounded = PropertyUnbounded.ARRAY,
+                value = {"integrations", "smtp", "email"},
+                description = "Tags for this check to be used by composite health checks."),
+        @Property(
+                name = HealthCheck.MBEAN_NAME,
+                value = "smtpMailService",
+                propertyPrivate = true)})
+@Service
+public class SMTPMailServiceHealthCheck implements HealthCheck {
+    private static final Logger log = LoggerFactory.getLogger(SMTPMailServiceHealthCheck.class);
+
+    private static String MAIL_TEMPLATE =
+            System.getProperty("line.separator") +
+            "Sling Health Check for AEM E-mail Service connectivity";
+
+    private static final String DEFAULT_EMAIL = "healthcheck@example.com";
+    @Property(
+            label = "Test E-mail Address",
+            description = "E-mail address to send test message to.",
+            value = DEFAULT_EMAIL)
+    private static final String PROP_EMAIL = "email";
+    private String email;
+
+    @Reference
+    private MessageGatewayService messageGatewayService;
+
+    @Activate
+    protected void activate(Map<String, Object> properties) {
+        this.email = PropertiesUtil.toString(properties.get(PROP_EMAIL), DEFAULT_EMAIL);
+    }
+
+    @Override
+    public Result execute() {
+        final FormattingResultLog resultLog = new FormattingResultLog();
+
+        if (messageGatewayService == null) {
+            resultLog.warn("MessageGatewayService OSGi service could not be found.");
+            resultLog.info("Verify the Default Mail Service is active: http://localhost:4502/system/console/components/com.day.cq.mailer.impl.CqMailingService");
+        } else {
+            final MessageGateway<SimpleEmail> messageGateway = messageGatewayService.getGateway(SimpleEmail.class);
+            if (messageGateway == null) {
+                resultLog.critical("The AEM Default Mail Service is INACTIVE, thus e-mails cannot be sent.");
+                resultLog.info("Verify the Default Mail Service is active and configured: http://localhost:4502/system/console/components/com.day.cq.mailer.DefaultMailService");
+                log.error("Could not retrieve a SimpleEmail Message Gateway");
+
+            } else {
+                try {
+                    List<InternetAddress> emailAddresses = new ArrayList<InternetAddress>();
+                    emailAddresses.add(new InternetAddress(this.email));
+                    MailTemplate mailTemplate = new MailTemplate(IOUtils.toInputStream(MAIL_TEMPLATE), CharEncoding.UTF_8);
+                    SimpleEmail email = mailTemplate.getEmail(StrLookup.mapLookup(new HashMap<String, String>()), SimpleEmail.class);
+
+                    email.setSubject("AEM E-mail Service Health Check");
+                    email.setTo(emailAddresses);
+
+                    try {
+                        messageGateway.send(email);
```

Comment:
Perhaps ignorance on my part, but is this a synchronous call, i.e. it will fail if the SMTP server is unreachable?